_layouts/resource.html

```diff
@@ -8,17 +8,35 @@
   {% assign type = "resource" %}
 {% endif %}
 
-{% if page.authors %}
-  {% assign authors = page.authors | join: ", " %}
+{% if page.author %}
+  {% assign resourceAuthors = nil %}
+  {% for author in page.author %}
+    {% assign name = nil %}
+    {% if author.name %}
+      {% assign name = author.name %}
+    {% elsif author.id %}
+      {% for person in site.data.people %}
+        {% if author.id == person.id %}
+          {% capture name %}{{ person.first}} {{person.last}}{% endcapture %}
+        {% endif %}
+      {% endfor %}
+    {% endif %}
+
+    {% if resourceAuthors != nil %}
+      {% assign addPerson =  name | prepend: ", " %}
+      {% assign resourceAuthors = resourceAuthors | append: addPerson %}
+    {% else %}
+      {% assign resourceAuthors = name %}
+    {% endif %}
+  {% endfor %}
 {% endif %}
```

Comment:
Wound up having to complicate this to allow the same style of author checks and overrides that the citations offer.